invenio_drafts_resources/services/records/components/files.py

```diff
@@ -9,50 +9,50 @@
 # details.
 
 """Records service component base classes."""
-
 from invenio_i18n import gettext as _
-from invenio_records_resources.services.files.transfer import TransferType
-from invenio_records_resources.services.records.components import FilesOptionsComponent
-from marshmallow import ValidationError
 
-from .base import ServiceComponent
+from .files_base import RecordFilesComponent
+from invenio_records_resources.services.records.components import (
+    MediaFilesOptionsComponent,
+)
+from marshmallow import ValidationError
 
 
-class DraftFilesComponent(ServiceComponent):
+class DraftFilesComponent(RecordFilesComponent):
     """Draft files service component."""
 
+    _files_attr_key = "files"
+    _files_data_key = "files"
+    _files_bucket_attr_key = "bucket"
+    _files_bucket_id_attr_key = "bucket_id"
+
+
+class DraftMediaFilesComponent(RecordFilesComponent):
+    """Draft media files component."""
+
+    _files_attr_key = "media_files"
+    _files_data_key = "media_files"
+    _files_bucket_attr_key = "media_bucket"
+    _files_bucket_id_attr_key = "media_bucket_id"
+
     def __init__(self, service, *args, **kwargs):
         """Constructor."""
         super().__init__(service)
-        self.files_component = FilesOptionsComponent(service)
+        self.files_component = MediaFilesOptionsComponent(service)
 
-    #
-    # API
-    #
     def create(self, identity, data=None, record=None, errors=None):
         """Assigns files.enabled.
 
         NOTE: `record` actually refers to the draft
               (this interface is used in records-resources and rdm-records)
         """
         draft = record
-        enabled = data.get("files", {}).get("enabled", True)
-
-        if draft.files.enabled != enabled:
-            if not self.service.check_permission(
-                identity, "manage_files", record=draft
-            ):
-                errors.append(
-                    {
-                        "field": "files.enabled",
-                        "messages": [
-                            _("You don't have permissions to manage files options.")
-                        ],
-                    }
-                )
-                return  # exit early
-
-        draft.files.enabled = enabled
+        files = self.get_record_files(draft)
+
+        # disable the media files by default
+        enabled = data.get(self.files_data_key, {}).get("enabled", False)
```

Comment:
in the media files we don't check the manage_files permissions, since media files are additional and they can be added without special restrictions